docs/docs/tutorials/orca-yarn-tutorial.md

```diff
@@ -0,0 +1,355 @@
+
+Also, a CDH cluster’s `HADOOP_CONF_DIR` should be `/etc/hadoop/conf` on CDH by default.
+
+# 3. Run Jobs on Yarn
+In the following part, we will show you three ways that BigDL supports to submit and run our programs on Yarn:
+* Use a built-in function
+* Use a BigDL provided script
+* Use a spark-submit script
+
+## 3.1 Run Programs with built-in function
+This is the easiest and most recommended way to run BigDL on YARN, you only need to prepare the environment on the driver machine, all dependencies would be automatically packaged and distributed to the whole Yarn cluster. In this way, you can easily switch your job between local (for test) and YARN (for production) by changing the `cluster_mode` to be yarn-client or yarn-cluster, `init_orca_context` will submit the job to YARN with client or cluster mode respectively.
+
+### 3.1.1 Yarn Client
+When running every Orca program on yarn-client mode using built-in function, we should call `init_orca_context` to create an OrcaContext at the very beginning of our programs.
+
+```python
+from bigdl.orca import init_orca_context
+
+init_orca_context(cluster_mode="yarn-client", cores=4, memory="10g", num_nodes=2)
+```
+
+To execute the tutorial provided Fashion-MINIST example, we should set `--cluster_mode` argument as `yarn-client` or `yarn` to create an OrcaContext and run the program on yarn-client mode as below:
+
+```bash
+python train.py --cluster_mode yarn-client
+```
+
+### 3.1.2 Yarn Cluster
+Fisrt, please call `init_orca_context` to create an OrcaContext at the very beginning of Orca programs. To run programs on yarn cluster mode using built-in function, we should switch the `cluster_mode` to `yarn-cluster` in `init_orca_context` as below:
+
+```python
+from bigdl.orca import init_orca_context
+
+init_orca_context(cluster_mode="yarn-cluster", cores=4, memory="10g", num_nodes=2)
+```
+
+When running the program provided by tutorial, we should:
+* Set `--cluster_mode` parameter to `yarn-cluster`, which could create an OrcaContext and run the Fasion-MNIST program on yarn-cluster mode.
+* Set `--remote_dir` parameter to the datasets path on remote resources. We recommand you to use datasets from remote resources like HDFS and S3 to instead of downloading these datasets, since there may get connection errors on `yarn-cluster` mode.
+
+Now, please run the example as the following script:
+```bash
+python train.py --cluster_mode yarn-cluster --remote_dir hdfs://path/to/remote/data
+```
+
+## 3.2 Run Programs with BigDL Provided Scripts
+For `spark-submit` users, BigDL provides a `bigdl-submit` script, you can find it in `${BIGDL_HOME}/scripts/bigdl-submit` and adjust the configurations according to your cluster settings. 
```

Comment:
talk about when and why should the user use `bigdl-submit` instead of `python`

Reply:
In the previous `init_orca_context` version, we don't support `init_on_yarn`, so `bigdl-submit` is the only choice. For now, they don't really have a difference as the reason that users should pick one.

Reply:
Then maybe there is no need to document `bigdl-dubmit`? And shall we add the description for using `jupyter` to run the job on yarn?

Reply:
I think we could keep the `bigdl-submit` option, in case some users may just want to use spark-submit and they are allowed to install python dependencies using conda, hence we could recommend them to use `bigdl-submit`. 

As for using `jupyter`, we could add the description. Should we also add a notebook example for `jupyter` option in this PR?

Reply:
> As for using `jupyter`, we could add the description. Should we also add a notebook example for `jupyter` option in this PR?

Is it similar to using `python`?

Reply:
> > As for using `jupyter`, we could add the description. Should we also add a notebook example for `jupyter` option in this PR?
> 
> Is it similar to using `python`?

Yes, quite similar.

Reply:
> > > As for using `jupyter`, we could add the description. Should we also add a notebook example for `jupyter` option in this PR?
> > 
> > 
> > Is it similar to using `python`?
> 
> Yes, quite similar.

Then maybe just briefly mention it at the end of the `python` section

Reply:
sure.

Reply:
> > > > As for using `jupyter`, we could add the description. Should we also add a notebook example for `jupyter` option in this PR?
> > > 
> > > 
> > > Is it similar to using `python`?
> > 
> > 
> > Yes, quite similar.
> 
> Then maybe just briefly mention it at the end of the `python` section

sure.